src/main/java/de/cytooxien/realms/api/PlayerInformationProvider.java

```diff
@@ -63,4 +64,11 @@ public interface PlayerInformationProvider {
      */
     Action<Boost[]> boosts(UUID playerId);
 
+    /**
+     *  This will return the JoinReason of the player
+     * @param playerId the player UUID to check
+     * @return the JoinReason
+     */
+    Action<JoinReason> joinReason(UUID playerId);
```

Comment:
That wouldn't be an action (no backend request). What should we do if the player is currently not online?

Reply:
Annotate with JetBrains Nullable Attribute?